### PR TITLE
File sharing feature

### PR DESCRIPTION
**Pull request by AntonGorielikov**
In work.

### CODE REVIEW

File: Chat/FileManager.cpp [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    for(File it : shared_files_)
+    {
+        return_vector.push_back(it.GetName());
+    }
+    return return_vector;
+}
+
+void FileManager::Save()
+{
+    if((CreateDirectory(FILE_DATA_SAVE_DIR, NULL)) ||
```
Comment (miklealex) [inline Chat/FileManager.cpp:58]: Pretty good.
Using filestream is not a big deal. 
We want you to use WINAPI not only in multythreading.
Please, use CreateFile, WriteFile, FILE functionality for work with files.

Comment (AntonGorielikov) [reply Chat/FileManager.cpp:58]: Ok, will rewrite it to winapi.

Comment (miklealex) [reply Chat/FileManager.cpp:58]: Thanks

File: Chat/Network.cpp [hunk 2]
```diff
@@ -145,6 +145,31 @@ void Network::SendFile(std::string pass, std::string ip)
 	 file_send_socket_.SendFile(pass, ip);
 }
 
+std::vector<std::string> Network::GetFileList()
```
Comment (miklealex) [inline Chat/Network.cpp]: I think, it is a little expensive to copy all vector. Maybe, there is the way to return reference?

File: Chat/Network.cpp [hunk 3]
```diff
@@ -145,6 +145,31 @@ void Network::SendFile(std::string pass, std::string ip)
...
+{
+    return FM_.GetFileNames();
+}
+
+void Network::RequestSomeoneList(std::string name)
```
Comment (miklealex) [inline Chat/Network.cpp]: maybe, const reference here and in line 162?

File: Chat/Network.cpp (hunk 2 above)
Comment (miklealex) [reply Chat/Network.cpp]: It is good practice to pass output variable as function parameter. I mean GetFileList(std::vector<std::string>& ).

File: Chat/Network.cpp (hunk 3 above)
Comment (AntonVeselskyi) [reply Chat/Network.cpp]: ok, will add const & everywhere if it`s possible.

Comment (miklealex) [reply Chat/Network.cpp]: Thanks

File: Chat/Network.cpp (hunk 2 above)
Comment (AntonVeselskyi) [reply Chat/Network.cpp]: Great idea, thank you

File: Chat/Chat.cpp [hunk 4]
```diff
@@ -96,7 +96,7 @@ void Chat::PutMsg(const UserMsg& msg)
 void Chat::PrintSomeoneList(std::vector<std::string>& list)
 {
     chat_mutex_.Lock();
-    for (int i = 0; i < list.size(); i++)
+    for (int i = 0; i < (int)list.size(); i++)
```
Comment (miklealex) [inline Chat/Chat.cpp]: why this conversion is needed? Is this because of warning about "size_t to int"?

File: Chat/FileManager.cpp [hunk 5]
```diff
@@ -37,60 +37,98 @@ std::string FileManager::GetFileName(int file_index)const
...
 
 void FileManager::Save()
 {
     if((CreateDirectory(FILE_DATA_SAVE_DIR, NULL)) ||
        (GetLastError() == ERROR_ALREADY_EXISTS))
     {
-        std::ofstream file_stream;
-        file_stream.open(FILE_DATA_SAVE_FULLPATH,
-                         std::ofstream::out);
+        HANDLE file = CreateFile(FILE_DATA_SAVE_FULLPATH,   //path
```
Comment (miklealex) [inline Chat/FileManager.cpp]: Nice one)

File: Chat/Chat.cpp (hunk 4 above)
Comment (AntonGorielikov) [reply Chat/Chat.cpp]: yea, just want to avoid warnings.
If i remember correctly it won't fail even without it.

Comment (miklealex) [reply Chat/Chat.cpp]: yeah, but you can create "size_t i" variable. Can you check if there will be some warnings in that case?
I mean "for(size_t i = 0; i <...."

Comment (AntonGorielikov) [reply Chat/Chat.cpp]: ok, will change it to this then

File: Chat/Thread.h [hunk 6]
```diff
@@ -10,7 +11,7 @@ class Thread
 {
 public:
     Thread();
-	Thread(thread_function func, void *params);
+    Thread(thread_function func, void *params);
```
Comment (miklealex) [inline Chat/Thread.h:14]: Please, use git diff to prevent commiting useless files.

Comment (AntonGorielikov) [reply Chat/Thread.h:14]: Ok. sorry.

Comment (miklealex) [reply Chat/Thread.h:14]: No problem

File: Chat/FileGetSocket.cpp [hunk 7]
```diff
@@ -77,6 +77,10 @@ bool FileGetSocket::GetFile()
                                    NULL);
         recv_size += recved_bytes;
         size += 1;
+        if (size > 2195)
```
Comment (miklealex) [inline Chat/FileGetSocket.cpp]: why such number?

**Pull request merged**